tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -94,7 +94,10 @@ private[scalanative] object LLVM {
         val compiler = if (isCpp) config.clangPP.abs else config.clang.abs
         val flags = (if (isCpp) Seq("-std=c++11")
                      else Seq("-std=gnu11")) ++ config.compileOptions
-        val compilec = Seq(compiler) ++ flags ++ Seq("-c", path, "-o", opath)
+        val compilec = Seq(compiler) ++ flto(config) ++ flags ++ Seq("-c",
```

Comment:
In order for this flag to automatically be picked up by projects that require compiling C/C++ code it would be better to make the LTO flag part of `config.compileOptions` and `compile. linkingOptions` and thus automatically exposing it in sbt and other build tools.

Coincidentally it would probably also make the PR simpler.

Reply:
This also coincides with some of the ideas here about some of the hardcoded values for debug and release as well. https://github.com/scala-native/scala-native/issues/1205

Reply:
Okay, makes sense. Then ideally this will be handled by #657.